local-cluster/src/cluster_tests.rs

```diff
@@ -527,20 +532,39 @@ pub fn start_gossip_voter(
                 });
 
                 for (parsed_vote, leader_vote_tx) in &parsed_vote_iter {
-                    if let Some(latest_vote_slot) = parsed_vote.last_voted_slot() {
+                    if let Some(vote_slot) = parsed_vote.last_voted_slot() {
+                        latest_vote_slot = vote_slot;
                         info!("received vote for {}", latest_vote_slot);
                         process_vote_tx(
                             latest_vote_slot,
                             leader_vote_tx,
                             parsed_vote,
-                            &cluster_info,
+                            &mut votes_to_sendout,
                         )
                     }
-                    // Give vote some time to propagate
-                    sleep(Duration::from_millis(sleep_ms));
                 }
 
-                if parsed_vote_iter.is_empty() {
+                // When push_active_set() is first called, Gossip doesn't have correct shred_version set, neither
+                // does it have any ping responses from peers, so messages pushed at this point will not be sent
+                // correctly. Wait until CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS / 2 + 1000 ms to give Gossip time to
+                // discover peers and set correct shred_version.
+                if timestamp() - start_time > (CRDS_GOSSIP_PULL_CRDS_TIMEOUT_MS / 2 + 1000)
```

Comment:
can we read the shred version directly from `cluster_info` to make sure it's correct rather than relying on time?

Reply:
Also seems like we could move these initialization checks directly at the beginning of `std::thread::spawn(move || {` before the thread loop starts